### ISSUE TITLE
Adding a Location via admin UI  causes an error when creating NewsItems

### ISSUE THREAD

**Issue by slinkp** [openplans/openblock#95]
It seems all you have to do is add a Location via the admin UI, and then try to add a NewsItem. When saving the NewsItem, you get this (note there are actually two errors here, one logging and one SQL):

```
2011-01-10 14:09:40,891 INFO     Starting add_events
Traceback (most recent call last):
  File "/usr/lib/python2.6/logging/__init__.py", line 768, in emit
    msg = self.format(record)
  File "/usr/lib/python2.6/logging/__init__.py", line 648, in format
    return fmt.format(record)
  File "/usr/lib/python2.6/logging/__init__.py", line 436, in format
    record.message = record.getMessage()
  File "/usr/lib/python2.6/logging/__init__.py", line 306, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
2011-01-10 14:09:43,567 ERROR    Traceback (most recent call last):
  File "scrapers/add_events.py", line 84, in main
    item.save()
  File "/home/pw/builds/openblock/builds/20101027/lib/python2.6/site-packages/django/db/models/base.py", line 434, in save
    self.save_base(using=using, force_insert=force_insert, force_update=force_update)
  File "/home/pw/builds/openblock/builds/20101027/lib/python2.6/site-packages/django/db/models/base.py", line 527, in save_base
    result = manager._insert(values, return_id=update_pk, using=using)
  File "/home/pw/builds/openblock/builds/20101027/lib/python2.6/site-packages/django/db/models/manager.py", line 195, in _insert
    return insert_query(self.model, values, **kwargs)
  File "/home/pw/builds/openblock/builds/20101027/lib/python2.6/site-packages/django/db/models/query.py", line 1479, in insert_query
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/home/pw/builds/openblock/builds/20101027/lib/python2.6/site-packages/django/db/models/sql/compiler.py", line 783, in execute_sql
    cursor = super(SQLInsertCompiler, self).execute_sql(None)
  File "/home/pw/builds/openblock/builds/20101027/lib/python2.6/site-packages/django/db/models/sql/compiler.py", line 727, in execute_sql
    cursor.execute(sql, params)
  File "/home/pw/builds/openblock/builds/20101027/lib/python2.6/site-packages/django/db/backends/util.py", line 15, in execute
    return self.cursor.execute(sql, params)
  File "/home/pw/builds/openblock/builds/20101027/lib/python2.6/site-packages/django/db/backends/postgresql_psycopg2/base.py", line 44, in execute
    return self.cursor.execute(query, args)
DatabaseError: Relate Operation called with a LWGEOMCOLLECTION type.  This is unsupported.
HINT:  Change argument 1: 'SRID=4326;GEOMETRYCOLLECTION(POLYGON((-71.088882751465 42.36838667362,-71.086...'
CONTEXT:  SQL statement "INSERT INTO db_newsitemlocation (news_item_id, location_id) SELECT  $1 , id FROM db_location WHERE intersects(db_location.location,  $2 )"
PL/pgSQL function "update_newsitem_location" line 27 at SQL statement
```

**Comment by slinkp** [comment]
Hello Guys!

Was wondering if there is any progress here? Without this getting fixed I am afraid I will not be able to add news items with locations...

Thanks.

**Comment by slinkp** [comment]
(In [bb93e5cf5a2722ba7726ddc88a207bcce8ca9e5e]) Fixes #95, fixes #74. Add a new signal and a handler for it that loads our custom location sql only after all the geom columns exist; use the intersecting_collection function in more places to avoid blowing up on GeometryCollections.

**Comment by slinkp** [comment]
Ticket imported from Trac:
 http://developer.openblockproject.org/ticket/95
Reported by: slinkp
